scipy/special/boost_special_functions.h

```diff
@@ -264,6 +264,48 @@ ibeta_inv_double(double a, double b, double p)
     return ibeta_inv_wrap(a, b, p, SpecialPolicy());
 }
 
+template<typename Real>
+static inline
+Real ibeta_inva_wrap(Real p, Real b, Real x)
+{
+    Real y;
+
+    if (std::isnan(p) || std::isnan(b) || std::isnan(x)) {
+        return NAN;
+    }
+    if ((b <= 0) || (x <= 0) || (p < 0) || (p > 1)) {
+        sf_error("btdtria", SF_ERROR_DOMAIN, NULL);
+        return NAN;
+    }
+    try {
+        y = boost::math::ibeta_inva(b, x, p, SpecialPolicy());
+    } catch (const std::domain_error& e) {
+        sf_error("btdtria", SF_ERROR_DOMAIN, NULL);
+        y = NAN;
+    } catch (const std::overflow_error& e) {
+        sf_error("btdtria", SF_ERROR_OVERFLOW, NULL);
+        y = INFINITY;
+    } catch (const std::underflow_error& e) {
+        sf_error("btdtria", SF_ERROR_UNDERFLOW, NULL);
+        y = 0;
+    } catch (...) {
+        sf_error("btdtria", SF_ERROR_OTHER, NULL);
+        y = NAN;
+    }
+    return y;
+}
+
+float
+ibeta_inva_float(float p, float b, float x)
```

Comment:
Does this templating mechanism handle `float`, `double`, `float` input combos?

Reply:
The ufunc machinery handles this. If there is no implementation for the given types, it searches for types the inputs can be safely cast to. See https://numpy.org/doc/stable/user/basics.ufuncs.html#type-casting-rules.